homeassistant/components/enocean/sensor.py

```diff
@@ -281,3 +292,21 @@ def value_changed(self, packet):
             self._attr_native_value = "tilt"
 
         self.schedule_update_ha_state()
+
+class EnOceanShutterContact(EnOceanSensor):
+    """Representation of an EnOcean shutter contact device.
+    
+    EEPs (EnOcean Equipment Profiles):
+    - D5-00-01 (Shutter contact)
+    """
+
+    def value_changed(self, packet):
+        """Update the internal state of the sensor."""
+        if packet.rorg != 0xD5:
```

Comment:
This integration has existing code that has too much protocol details. We won't allow new code that introduces more protocol details. For more sensor types to be added we need to extract the device specific interface code and data parsing to a standalone library published on PyPI.

Reply:
I just now saw your comment in another PR adding the same feature... Should have looked into the closed ones as well. I see your point to not include protocol specific code, but it is quite bad we cannot integrate any non-existing enocean hardware for now. Do you know if someone is already working on a new library? [The one we are using right now](https://github.com/kipe/enocean) seems to be abandoned anyway.

Reply:
I don't know of any progress in this direction lately, unfortunately. There's an Enocean Discord server for talking about the integration but there's not much activity.